lib/web_ui/lib/src/engine/window.dart

```diff
@@ -643,20 +643,20 @@ class EngineWindow extends ui.Window {
       case 'flutter/navigation':
         const MethodCodec codec = JSONMethodCodec();
         final MethodCall decoded = codec.decodeMethodCall(data);
-        final Map<String, dynamic>? message = decoded.arguments;
+        final Map<String, dynamic> message = decoded.arguments ?? const <String, dynamic>{};
```

Comment:
@mdebbar Not sure if I should just do `as Map<String, dynamic>` instead. I imagine we always get a map?

Reply:
Today, yes we always get a map, but I'm not sure if we can safely make assumptions like that with platform messages since there's no type-checking whatsoever.

cc @chunhtai

Reply:
We already cast to `Map?` today. My code isn't changing that part. What I'm wondering is whether we should cast to `Map` instead. In the switch statement, we're null-asserting, so we're not expecting a null value.

Reply:
For example, is it possible to get a method call where `decoded.method` is neither `routeUpdated` nor `routeInformationUpdated`, perhaps just to reset `_defaultRouteName`. That would be the only situation I can think of where `message` would be null and we wouldn't crash.

Reply:
We could add new methods in the future that don't conform to this contract. But in that case, even `Map?` may fail.

TL;DR: go ahead and cast to `Map`.

Reply:
I meant that the message for the new method could be anything. It could be a list, right? Or a number, string, etc.

Reply:
I think it is ok to cast it `Map`. We can always change that if we have new contract in the future.

Reply:
Cool. Casting to `Map` then.